src/stream/src/executor/backfill.rs

```diff
@@ -381,7 +381,12 @@ where
 
     fn mapping_message(msg: Message, upstream_indices: &[usize]) -> Message {
         match msg {
-            Message::Barrier(_) | Message::Watermark(_) => msg,
+            Message::Barrier(_) => msg,
+            Message::Watermark(watermark) => Message::Watermark(Watermark {
+                col_idx: upstream_indices[watermark.col_idx],
```

Comment:
The `upstream_indices` seems to be a projection, that is, we should find the position of the `col_idx` value and ignore the message if not exists.

Reply:
Oops! what a mistake.